types/struct.go

```diff
@@ -240,6 +241,9 @@ func (b *structSSZ) Unmarshal(val reflect.Value, typ reflect.Type, input []byte,
 				continue
 			}
 			nextOff := offsets[offsetIndex+1]
+			if nextOff >= uint64(len(input)) {
```

Comment:
If the last element is dynamic and has a 0 size it can be equal to the scope end